### PR TITLE
chore: Disallow using a key before it is created: based on update property

### PR DESCRIPTION
**Pull request by ukstv**

### CODE REVIEW

File: src/did.ts [hunk 1]
```diff
@@ -216,11 +216,19 @@ export class DID {
...
+      if (isLater) {
         // Do not allow using a key _after_ it is being revoked
         throw new Error(`JWS was signed with a revoked DID version: ${kid}`)
       }
     }
+    // Key used before `updated` date
+    const updated = didResolutionResult.didDocumentMetadata?.updated
+    const versionId = didResolutionResult.didDocumentMetadata?.versionId
+    const notV0 = versionId !== '0'
+    if (notV0 && updated && options.atTime && options.atTime < new Date(updated).valueOf()) {
```
Comment (oed) [inline src/did.ts:229]: Were we not going to remove the versionId check here, as it's 3ID specific?

Comment (oed) [reply src/did.ts:229]: Also what happens if updated is not defined here. Can be the case for various DID methods, e.g. Key DID.

Comment (ukstv) [reply src/did.ts:229]: Ooh. Stupid me. Removed that in https://github.com/ceramicnetwork/js-did/pull/40

Comment (ukstv) [reply src/did.ts:229]: > Also what happens if updated is not defined here. Can be the case for various DID methods, e.g. Key DID.

No `updated` ⇒ no reference timestamp ⇒ do not throw.

Comment (oed) [reply src/did.ts:229]: I see, silly me :P

**Comment by ukstv** [comment]
So @oed is it a formal yes to merge both PRs?

**Review by oed — APPROVED** [review]
Approved on the condition that #40 get's merged ;)

**Pull request merged**